diff --git a/model/ModelBarcodeObject.hpp b/model/ModelBarcodeObject.hpp
--- a/model/ModelBarcodeObject.hpp
+++ b/model/ModelBarcodeObject.hpp
@@ -80,6 +80,111 @@ namespace glabels::model
                 bool bcTextFlag() const override;
                 void setBcTextFlag( bool value ) override;
 
+                //
+                // Human-readable barcode text font
+                //
+                QString fontFamily() const override
+                {
+                        return mTextFontFamily;
+                }
+
+                void setFontFamily( const QString& value ) override
+                {
+                        if ( mTextFontFamily != value )
+                        {
+                                mTextFontFamily = value;
+                                emit changed();
+                        }
+                }
+
+                double fontSize() const override
+                {
+                        return mTextFontSize;
+                }
+
+                void setFontSize( double value ) override
+                {
+                        if ( mTextFontSize != value )
+                        {
+                                mTextFontSize = value;
+                                emit changed();
+                        }
+                }
+
+                QFont::Weight fontWeight() const override
+                {
+                        return mTextFontWeight;
+                }
+
+                void setFontWeight( QFont::Weight value ) override
+                {
+                        if ( mTextFontWeight != value )
+                        {
+                                mTextFontWeight = value;
+                                emit changed();
+                        }
+                }
+
+                bool fontItalicFlag() const override
+                {
+                        return mTextFontItalicFlag;
+                }
+
+                void setFontItalicFlag( bool value ) override
+                {
+                        if ( mTextFontItalicFlag != value )
+                        {
+                                mTextFontItalicFlag = value;
+                                emit changed();
+                        }
+                }
+
+                bool fontUnderlineFlag() const override
+                {
+                        return mTextFontUnderlineFlag;
+                }
+
+                void setFontUnderlineFlag( bool value ) override
+                {
+                        if ( mTextFontUnderlineFlag != value )
+                        {
+                                mTextFontUnderlineFlag = value;
+                                emit changed();
+                        }
+                }
+
+                Qt::Alignment textHAlign() const override
+                {
+                        return mTextHAlign;
+                }
+
+                void setTextHAlign( Qt::Alignment value ) override
+                {
+                        if ( mTextHAlign != value )
+                        {
+                                mTextHAlign = value;
+                                emit changed();
+                        }
+                }
+
+                //
+                // Human-readable barcode text distance
+                //
+                double bcTextGap() const
+                {
+                        return mTextGap;
+                }
+
+                void setBcTextGap( double value )
+                {
+                        if ( mTextGap != value )
+                        {
+                                mTextGap = value;
+                                emit changed();
+                        }
+                }
+
+
 
                 //
                 // Barcode Property: bcChecksumFlag
@@ -155,4 +260,12 @@ namespace glabels::model
                 std::unique_ptr<glbarcode::Barcode> mEditorBarcode;
                 std::unique_ptr<glbarcode::Barcode> mEditorDefaultBarcode;
 
+                QString             mTextFontFamily{ "Sans" };
+                double              mTextFontSize{ 10.0 };
+                QFont::Weight       mTextFontWeight{ QFont::Normal };
+                bool                mTextFontItalicFlag{ false };
+                bool                mTextFontUnderlineFlag{ false };
+                Qt::Alignment       mTextHAlign{ Qt::AlignHCenter };
+                double              mTextGap{ 1.0 };
+
                 QPainterPath mHoverPath;
