diff --git a/glbarcode/QtRenderer.hpp b/glbarcode/QtRenderer.hpp
--- a/glbarcode/QtRenderer.hpp
+++ b/glbarcode/QtRenderer.hpp
@@ -52,14 +52,15 @@
                 /** Get "painter" parameter
                  *
                  * @returns painter parameter
                  */
                 QPainter* painter() const;
                 /** Set "painter" parameter
                  *
                  * @param[in] painter pointer to QPainter
                  *
                  * @returns reference to this QtRenderer object for parameter chaining
                  */
                 QtRenderer& setPainter( QPainter* painter );
+                QtRenderer& setRenderText( bool renderText );

         private:
diff --git a/glbarcode/QtRenderer.cpp b/glbarcode/QtRenderer.cpp
--- a/glbarcode/QtRenderer.cpp
+++ b/glbarcode/QtRenderer.cpp
@@ -37,6 +37,7 @@
         struct QtRenderer::PrivateData
         {
                 QPainter* painter;
                 QColor    color;
+                bool      renderText{true};
         };


@@ -65,14 +66,21 @@
         QPainter* QtRenderer::painter( ) const
         {
                 return d->painter;
         }


         QtRenderer& QtRenderer::setPainter( QPainter* painter )
         {
                 d->painter = painter;
                 return *this;
         }


+        QtRenderer& QtRenderer::setRenderText( bool renderText )
+        {
+                d->renderText = renderText;
+                return *this;
+        }
+
+
         void QtRenderer::drawBegin( double w, double h )
@@ -118,7 +126,7 @@
         void QtRenderer::drawText( double x, double y, double size, const std::string& text, HAlign halign )
         {
-                if ( d->painter )
+                if ( d->painter && d->renderText )
                 {
                         d->painter->setPen( QPen( d->color ) );

                         QFont font;
