diff --git a/glbarcode/QtRenderer.hpp b/glbarcode/QtRenderer.hpp
--- a/glbarcode/QtRenderer.hpp
+++ b/glbarcode/QtRenderer.hpp
@@ -56,10 +56,11 @@
                 QPainter* painter() const;
                 /** Set "painter" parameter
                  *
                  * @param[in] painter pointer to QPainter
                  *
                  * @returns reference to this QtRenderer object for parameter chaining
                  */
                 QtRenderer& setPainter( QPainter* painter );
+                QtRenderer& setRenderText( bool renderText );
 
         private:
diff --git a/glbarcode/QtRenderer.cpp b/glbarcode/QtRenderer.cpp
--- a/glbarcode/QtRenderer.cpp
+++ b/glbarcode/QtRenderer.cpp
@@ -37,5 +37,6 @@
         struct QtRenderer::PrivateData
         {
                 QPainter* painter;
                 QColor    color;
+                bool      renderText{true};
         };
@@ -71,8 +72,15 @@
         QtRenderer& QtRenderer::setPainter( QPainter* painter )
         {
                 d->painter = painter;
                 return *this;
         }
 
 
+        QtRenderer& QtRenderer::setRenderText( bool renderText )
+        {
+                d->renderText = renderText;
+                return *this;
+        }
+
+
         void QtRenderer::drawBegin( double w, double h )
@@ -118,4 +126,4 @@
         void QtRenderer::drawText( double x, double y, double size, const std::string& text, HAlign halign )
         {
-                if ( d->painter )
+                if ( d->painter && d->renderText )
                 {
